lib/reporting/fraud_metrics_lg99_report.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+require 'csv'
+begin
+  require 'reporting/cloudwatch_client'
+  require 'reporting/cloudwatch_query_quoting'
+  require 'reporting/command_line_options'
+rescue LoadError => e
+  warn 'could not load paths, try running with "bundle exec rails runner"'
+  raise e
+end
+
+module Reporting
+  class FraudMetricsLg99Report
+    include Reporting::CloudwatchQueryQuoting
+
+    attr_reader :time_range
+
+    module Events
+      IDV_PLEASE_CALL_VISITED = 'IdV: Verify please call visited'
+      IDV_SETUP_ERROR_VISITED = 'IdV: Verify setup errors visited'
+
+      def self.all_events
+        constants.map { |c| const_get(c) }
+      end
+    end
+
+    # @param [Range<Time>] time_range
+    def initialize(
+      time_range:,
+      verbose: false,
+      progress: false,
+      slice: 3.hours,
+      threads: 5
+    )
+      @time_range = time_range
+      @verbose = verbose
+      @progress = progress
+      @slice = slice
+      @threads = threads
+    end
+
+    def verbose?
+      @verbose
+    end
+
+    def progress?
+      @progress
+    end
+
+    def as_emailable_reports
+      Reporting::EmailableReport.new(
+        title: 'LG-99 Metrics',
+        table: lg99_metrics_table,
+        filename: 'lg99_metrics',
+      )
+    end
+
+    def lg99_metrics_table
+      [
+        ['Metric', 'Total'],
+        ['Unique users seeing LG-99', totals('unique_users_count')],
+      ]
+    rescue Aws::CloudWatchLogs::Errors::ThrottlingException => err
+      [
+        ['Error', 'Message'],
+        [err.class.name, err.message],
+      ]
+    end
+
+    def to_csv
+      CSV.generate do |csv|
+        lg99_metrics_table.each do |row|
+          csv << row
+        end
+      end
+    end
+
+    # @return Array<Hash>
+    def data
+      @data ||= begin
+        fetch_results
+      end
+    end
+
+    def fetch_results
+      cloudwatch_client.fetch(query:, from: time_range.begin, to: time_range.end)
+    end
+
+    def query
+      params = {
+        event_names: quote(Events.all_events),
+      }
+
+      format(<<~QUERY, params)
+        fields
+            name
+          , @timestamp
+          , properties.user_id as user_id,
+          , properties.new_event AS new_event
+        | filter properties.new_event = 1
+        | filter name in %{event_names}
+        | stats count_distinct(user_id) as `unique_users_count`
```

Comment:
similarly, `count_distinct` is known to be inaccurate, we should do a count by user ID and unique in ruby, or just do it all in ruby

> Returns the number of unique values for the field. If the field has very high cardinality (contains many unique values), the value returned by count_distinct is just an approximation.

[source](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/CWL_QuerySyntax-Stats.html#:~:text=Returns%20the%20number%20of%20unique%20values%20for%20the%20field.%20If%20the%20field%20has%20very%20high%20cardinality%20(contains%20many%20unique%20values)%2C%20the%20value%20returned%20by%20count_distinct%20is%20just%20an%20approximation.)

Reply:
Ahh! i was doing like this way as we did for proofing rate report, but then i implemented as we are doing for MFA report. I will change.

Reply:
none of our reports should be using `count_distinct` ... where did we find that as a reference? the only one I can find is our manual reimplementation in `query_cloudwatch.rb`